src/api/profile.ts

```diff
@@ -38,3 +39,37 @@ export async function deleteProfile(opts?: AxiosRequestConfig) {
 
   return deleteOK;
 }
+
+/**
+ * Function to get customer account ID based on phone number
+ *
+ * @param phoneNumber phone number with prefix
+ * @returns {string | GetAccountByPhoneErrorCode}
+ *  - customer_account_id as string if successful,
+ *  - "no_associated_account" if account ID not found,
+ *  - "unknown_error" if there's other error
+ *
+ */
+export const getCustomerAccountId = async (
+  phoneNumber: string,
+): Promise<string | GetAccountByPhoneErrorCode> => {
+  return await client
+    .post(
+      `${profileEndpoint}/search`,
+      {phoneNumber: phoneNumber},
+      {
+        authWithIdToken: true,
+        skipErrorLogging: (error) =>
+          error.response?.status === 404 || error.response?.status === 400,
+      },
+    )
+    .then((response) => response.data.customerAccountId as string)
+    .catch((error) => {
+      const metadata = getAxiosErrorMetadata(error);
+      const responseStatus = metadata.responseStatus;
+      if (responseStatus === 404 || responseStatus === 400) {
+        return 'no_associated_account';
+      }
```

Comment:
I would still prefer returning `undefined` here, as account not found is a normal situation, and not considered an error.

I think this touches on this old school principle:
https://web.archive.org/web/20140430044213/http://c2.com/cgi-bin/wiki?DontUseExceptionsForFlowControl

Reply:
Updated!